### ISSUE TITLE
Any plans for gradient checkpointing?

### ISSUE THREAD

**Issue by fred-hz** [allenai/allennlp#4211]
Do we have any future plans for supporting gradient checkpointing?

Like what pytorch does here. https://pytorch.org/docs/stable/checkpoint.html

Or tianqi's original paper: https://arxiv.org/abs/1604.06174

I think this will significantly decrease the GPU memory needed for models, especially when I want to train bert-large on my 1080ti...

**Comment by matt-gardner** [comment]
@fred-hz, from what I can tell, that kind of gradient checkpointing is implemented just as part of your `forward` implementation.  That means you can use it in your model code wherever you want, and it shouldn't require any special support from us.  Am I missing something here?

**Comment by fred-hz** [comment]
> @fred-hz, from what I can tell, that kind of gradient checkpointing is implemented just as part of your `forward` implementation. That means you can use it in your model code wherever you want, and it shouldn't require any special support from us. Am I missing something here?

I think you're correct. Thanks for your reply!
